lib/sla.js

```diff
@@ -189,96 +243,276 @@ function originateChannel(client, inbound, bridge, trunkData) {
  * @return {Q} - Q promise object
  */
-function getTrunkData (configurationFile, extension) {
-  return dal.getSharedExtension(configurationFile, extension)
-    .then(function(sharedExtension) {
-      return sharedExtension.getTrunk(extension).then(function(trunk) {
-      return Q.resolve(trunk);
-    })
-    .catch(function(invalid) {
-      // This will only be called if trunk and extension are not the same
-      // Shouldn't be right now
-      return Q.reject(new customError.CustomError('InvalidTrunkSpecification',
-          invalid));
+function isStation (extension, channel) {
+  return extension.getAllStations().then(function(stations) {
+    var isStation = false;
+    stations.some(function(station) {
+      var re = regexp().find(station).toRegExp();
+      // First conditional is for regular SIP users like SIP/phone1
+      if (re.test(channel.name)) {
+        isStation = true;
+        return true; 
+      } else {
+        // Second conditional is for other SIP users that have a different
+        // server like SIP/phone2@1234
+        var nameDivided = channel.name.split('/');
+        if (re.test(nameDivided[0] + '/' + channel.caller.number + '@' +
+            nameDivided[1])) {
+              isStation = true;
+              return true;
+            }
+      }
     });
-  })
-  .catch(function(invalid) {
-    if (invalid.name === 'InvalidSpecification') {
-      return Q.reject(invalid);
-    } else {
-      return Q.reject(new customError.CustomError('InvalidConfiguration',
-          invalid.message));
-    }
+    return Q.resolve(isStation);
   });
 }
 
+
+/**
+ * Returns the trunkData related to the extension in question
+ * @param {Object} extension - the extension object which has both the stations
+ *   and trunks
+ * @return {Array} - the array of trunk endpoint strings
+ */
+function getTrunkData (extension) {
+  return extension.getAllTrunks();
+}
+
 /**
  * Receives initial input and begins the application.
  * @param {Object} client - Client received from app.js.
  * @param {string} confFile - the configuration path and file name
- * @param {Object} inbound - the inbound channel
- * @param {string} extension - the name of the extension to access
+ * @param {Object} channel - the inbound or outbound channel
+ * @param {string} extensionName - the name of the extension to access
  * @return {Q} Q promise object.
  */
-module.exports = function(client, confFile, inbound, extension) {
-  return getTrunkData(confFile, extension)
-    .then(function(trunk) {
-      var answer = Q.denodeify(inbound.answer.bind(inbound));
-      return answer()
-        .then(function() {
-          return designateMixingBridge(client, extension);
-        }).then(function (bridge) {
-          return originateChannel(client, inbound, bridge, trunk)
+module.exports = function(client, confFile, channel, extensionName) {
+  return dal.getSharedExtension(confFile, extensionName)
+    .then(function(extension) {
```

Comment:
This is getting really unwieldy. I would do some research on ways to limit nested promises.